src/core/tests/Test_CoreMain.py

```diff
@@ -1147,23 +1147,29 @@ def test_temp_folder_created_during_execution_config_init(self):
 
         # temp_folder is set to None in ExecutionConfig with an invalid config_folder location, throws exception
         argument_composer = ArgumentComposer()
-        shutil.rmtree(argument_composer.temp_folder)
+        shutil.rmtree(str(argument_composer.temp_folder))
         argument_composer.temp_folder = None
         argument_composer.operation = Constants.ASSESSMENT
         # mock path exists check to return False on config_folder exists check
         backup_os_path_exists = os.path.exists
         os.path.exists = self.mock_os_path_exists
         self.assertRaises(Exception, lambda: RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT))
         # validate temp_folder is not created
-        self.assertFalse(os.path.exists(os.path.join(os.path.curdir, "scratch", "tmp")))
         os.path.exists = backup_os_path_exists
+        self.assertFalse(os.path.exists(os.path.join(os.path.curdir, "scratch", "tmp")))
```

Comment:
@rane-rajasi - the previous ordering of these lines made the assertion always False (false-confidence test).

Reply:
That is the intention of this test right? see comment on line 1153: # mock path exists check to return False on config_folder exists check